### PR TITLE
[Doc] Add 3.1 release notes

### PR DESCRIPTION
**Pull request by EsoragotoSpirit**
Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>

Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: docs/release_notes/release-3.1.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+- Supports the [AUTO_INCREMENT](../sql-reference/sql-statements/auto_increment.md) column attribute, which enables a globally unique ID for each data row and thus simplifies data management.
+- Supports [automatically creating partitions during loading and using partitioning expressions to define partitioning rules](../table_design/automatic_partitioning.md), thereby making partition creation easier to use and more flexible.
+- [Preview] Supports storing data on Azure Blob Storage.
+<!--- Supports abstraction of storage volumes, which makes it easier for users to configure storage location and authentication information in StarRocks shared-data clusters.-->
+
+#### Data Lake analytics
+
+- Supports accessing Parquet-formatted Iceberg v2 tables.
+- [Preview] Supports sinking data to Iceberg tables in Parquet format.
+- Supports accessing data stored in Elasticsearch by using [Elasticsearch catalogs](../data_source/catalog/elasticsearch_catalog.md). This simplifies the creation of Elasticsearch tables.
```
Comment (evelynzhaojie) [inline docs/release_notes/release-3.1.md:21]: 外表 This simplifies the creation of Elasticsearch external tables.

File: docs/release_notes/release-3.1.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+#### Storage engine, data ingestion, and query
+
+<!--- Supports [list partitioning](../table_design/).-->
+- Supports [random bucketing](../sql-reference/sql-statements/data-definition/CREATE%20TABLE.md#distribution_desc), which relieves the need to configure bucketing columns at table creation. In big data and high performance-demanding scenarios, we recommend that you continue using hash bucketing.
+- Supports using the TABLE keyword in [INSERT INTO](../loading/InsertInto.md) to directly load the data of Parquet- or ORC-formatted data files stored in AWS S3.
+- Supports [generated columns](../sql-reference/sql-statements/generated_columns.md). With the generated column feature, StarRocks can automatically generate and store the values of column expressions and automatically rewrite queries to improve query performance.
+- Supports loading data into columns of the [MAP](../sql-reference/sql-statements/data-types/Map.md) and [STRUCT](../sql-reference/sql-statements/data-types/STRUCT.md) data types, and supports nesting the Fast Decimal data type in the ARRAY, MAP, and STRUCT data types.
```
Comment (evelynzhaojie) [inline docs/release_notes/release-3.1.md:29]: and supports nesting Fast Decimal values in ARRAY, MAP, and STRUCT.

File: docs/release_notes/release-3.1.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  - Supports more expressions, such as `CASE-WHEN`, `CAST`, and mathematical operations, which make materialized views suitable in more business scenarios.
+
+#### Data Lake analytics
+
+- Optimized metadata caching and access for Iceberg to improve Iceberg data query performance.
+- Optimized the data cache to further improve data lake analytics performance.
+
+#### Storage engine, data ingestion, and query
+
+- Supports partial updates in column mode. Users can enable the column mode when they perform partial updates on Primary Key tables by using the UPDATE statement. The column mode is suitable for updating a small number of columns but a large number of rows, and can improve the updating performance by up to 10 times.
```
Comment (evelynzhaojie) [inline docs/release_notes/release-3.1.md:81]: add a doc link

File: docs/release_notes/release-3.1.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  - Supports using the `REFRESH MATERIALIZED VIEW WITH SYNC MODE` syntax to synchronously invoke materialized view refresh tasks.
+- Enhanced the use of asynchronous materialized views:
+  - Supports using `ALTER MATERIALIZED VIEW {ACTIVE | INACTIVE}` to enable or disable a materialized view. Materialized views that are disabled (in the `INACTIVE` state) cannot be refreshed or used for query rewrite, but can be directly queried.
+  - Supports using `ALTER MATERIALIZED VIEW SWAP WITH` to replace a materialized view. Users can create a new materialized view and then perform an atomic replacement on an existing materialized view to implement schema changes on the existing materialized view.
+- Optimized synchronous materialized views:
+  - Supports direct queries against synchronous materialized views using SQL hints `[_SYNC_MV_]`, allowing for walking around issues that some queries cannot be properly rewritten in rare circumstances.
+  - Supports more expressions, such as `CASE-WHEN`, `CAST`, and mathematical operations, which make materialized views suitable in more business scenarios.
```
Comment (evelynzhaojie) [inline docs/release_notes/release-3.1.md:72]: suitable in --- suitable for

File: docs/release_notes/release-3.1.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+  - Optimized the mechanism for rewriting queries that contain union operations, and supports rewriting queries that contain joins or functions such as COUNT DISTINCT and time_slice.
+- Optimized the refreshing of asynchronous materialized views:
+  - Optimized the mechanism for refreshing materialized views that are created on Hive catalog tables. StarRocks now can perceive partition-level data changes, and refreshes only the partitions with data changes during each automatic refresh.
+  - Supports using the `REFRESH MATERIALIZED VIEW WITH SYNC MODE` syntax to synchronously invoke materialized view refresh tasks.
+- Enhanced the use of asynchronous materialized views:
+  - Supports using `ALTER MATERIALIZED VIEW {ACTIVE | INACTIVE}` to enable or disable a materialized view. Materialized views that are disabled (in the `INACTIVE` state) cannot be refreshed or used for query rewrite, but can be directly queried.
+  - Supports using `ALTER MATERIALIZED VIEW SWAP WITH` to replace a materialized view. Users can create a new materialized view and then perform an atomic replacement on an existing materialized view to implement schema changes on the existing materialized view.
```
Comment (evelynzhaojie) [inline docs/release_notes/release-3.1.md:69]: atomic replacement, atomic exchange, atomic swap in release notes and docs --- suggest using consistent terms

**Review by hellolilyliuyi — APPROVED** [review]

**Review by evelynzhaojie — APPROVED** [review]

**Pull request merged**